### PR TITLE
disable audio on lost focus

### PR DESCRIPTION
**Pull request by Laserlicht**
HD feature.

For debugging it's helpful (if you have HD and VCMI opened the same time).

Do we need this also in regular codebase?

### CODE REVIEW

**Comment by Nordsoft91** [comment]
Couple ideas:
 - play sound if chat message received even in the background
 - play sound if turn received even in the background

**Comment by Laserlicht** [comment]
@Nordsoft91 Only disable music?

**Comment by Nordsoft91** [comment]
I think it's more tricky. There are environment sounds you may want to not have but want to receive sound notifications when have turn or message.

**Comment by Laserlicht** [comment]
Like this?

**Comment by Nordsoft91** [comment]
Yes. Make sense for new day/new week sounds too

File: client/adventureMap/CInfoBar.cpp [hunk 1]
```diff
@@ -228,14 +228,22 @@ CInfoBar::VisibleComponentInfo::VisibleComponentInfo(const std::vector<Component
...
+		handle = CCS->soundh->playSound(soundBase::newWeek);
 	else if(LOCPLINT->cb->getDate(Date::MONTH) != 1) // not first month
-		CCS->soundh->playSound(soundBase::newMonth);
+		handle = CCS->soundh->playSound(soundBase::newMonth);
 	else
-		CCS->soundh->playSound(soundBase::newDay);
+		handle = CCS->soundh->playSound(soundBase::newDay);
+
+	if(volume == 0)
+		CCS->soundh->setCallback(handle, [&]() { CCS->soundh->setVolume(0); });
```
Comment (IvanSavenko) [inline client/adventureMap/CInfoBar.cpp:246]: If player refocuses game when sound is playing he'll end up without sound

**Review by IvanSavenko — APPROVED** [review]

**Pull request merged**